core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -104,7 +104,7 @@ private[spark] class Executor(
       .setNameFormat("Executor task launch worker-%d")
       .setThreadFactory((r: Runnable) => new UninterruptibleThread(r, "unused"))
       .build()
-    Executors.newCachedThreadPool(threadFactory).asInstanceOf[ThreadPoolExecutor]
+    ThreadUtils.newCachedThreadPool(threadFactory)
```

Comment:
Do we need this change? This thread pool is used to execute `TaskRunner`,  which already sets MDC by itself, in https://github.com/apache/spark/pull/26624/files#diff-5a0de266c82b95adb47d9bca714e1f1bR380

Reply:
1. I think in any case it better to reuse the same code in all places.
2. in case later someone will add MDC set in the caller thread it will not pass without that.
so I think it good to use it here.

Reply:
> in case later someone will add MDC set in the caller thread it will not pass without that

It won't work anyway, as we overwrite the MDC properties in `TaskRunner.run`.

Reply:
makes sense